app/src/main/java/com/quran/labs/androidquran/widgets/HighlightingImageView.java

```diff
@@ -130,17 +152,178 @@ public void setNightMode(boolean isNightMode, int textBrightness) {
     adjustNightMode();
   }
 
-  public void highlightAyah(int sura, int ayah, HighlightType type) {
+  private void highlightFloatableAyah(Set<String> highlights, String currentSurahAyah) {
+    final Map<String, List<AyahBounds>> coordinatesData = ayahCoordinates.getAyahCoordinates();
+
+    String previousSurahAyah = currentSurahAyah;
+    for(String surahAyahs: highlights) {
+      previousSurahAyah = surahAyahs;
+    }
+    List<AyahBounds> startingBounds;
+    int arrowIndex = previousSurahAyah.indexOf("->");
+    if(arrowIndex != -1) {
+      // The ayah changed during animating
+      startingBounds = (List<AyahBounds>)animator.getAnimatedValue();
+      animator.cancel();
+      previousSurahAyah = previousSurahAyah.substring(arrowIndex + 2);
+    } else {
+      startingBounds = coordinatesData.get(previousSurahAyah);
+    }
+
+    final String ayahTransition = previousSurahAyah+"->"+currentSurahAyah;
```

Comment:
i'd vote for an object - maybe `AyahTransition` or something that has a from and to instead of manually having an arrow and parsing it like this, since it's more complicated that way

Reply:
Have created a class `AyahTransition` that handles this behaviour. Looking at your comment, I am getting an impression that you wanted me to remove the arrow logic altogether, but unfortunately, since we are using HashMap of Strings, I need some key to signify the transition, and the arrow is helpful in that case.